trollpy/tests.py

```diff
@@ -0,0 +1,357 @@
+    """Test that add_smack route shows form."""
+    testapp.post('/login',
+                 params={'username': 'amos',
+                         'password': 'password'})
+    response = testapp.get('/add_smack', status=200)
+    assert len(response.html.find_all('form')) == 1
+
+
+def test_add_smack_with_no_user(testapp, fill_the_db):
+    """Test that add_smack route shows 403."""
+    assert testapp.get('/add_smack', status=403)
+
+
+def test_add_smack_post(testapp, fill_the_db):
+    """Test adding smack to db."""
+    testapp.post('/login',
+                 params={'username': 'amos',
+                         'password': 'password'})
+    kill_score = {'killscore_id': -1, 'statement': 'aosidnaosidj'}
+    csrf = testapp.get('/add_smack').html.find_all("input")
+    csrf = csrf[0].attrs['value']
+    testapp.post('/add_smack', params={
+                 'killscore_id': -1,
+                 'statement': 'aosidnaosidj',
+                 'csrf_token': csrf}, status=302)
+    assrtion = testapp.get('/smack_api', status=200)
+    assert kill_score['statement'] == assrtion.json[-1]['statement']
+
+
+def test_del_smack(testapp, fill_the_db):
+    """Delete smack talk route."""
+    testapp.post('/login',
+                 params={'username': 'amos',
+                         'password': 'password'})
+    csrf = testapp.get('/add_smack').html.find_all("input")
+    csrf = csrf[0].attrs['value']
+    testapp.post('/add_smack', params={
+                 'killscore_id': -1,
+                 'statement': 'aosidnaosidj',
+                 'csrf_token': csrf}, status=302)
+    testapp.get('/delete/1', params={
+                'csrf_token': csrf}, status=302)
+    assrtion = testapp.get('/smack_api', status=200)
+    assert assrtion.json[0]['id'] != 1
+
+
+def test_user_board_json(testapp, fill_the_db):
+    """Test that get user board gets current chess board."""
+    board = testapp.get('/amos/api', status=200).json['board']
+    assert board == 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1'
+
+
+def test_make_move(testapp, fill_the_db):
+    """Test that making a move adds new board to the db."""
+    board = 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1'
+    move = 'rnbqkbnr/pppppppp/8/8/8/2N5/PPPPPPPP/R1BQKBNR b KQkq - 1 1'
+    testapp.post('/amos/move', {'board': move}, status=302)
+    board_sql = testapp.get('/amos/api', status=200).json['board']
+    assert not board == board_sql
+
+
+def test_user_list(testapp, fill_the_db):
+    """Test user list renders users."""
+    testapp.post('/login',
+                 params={'username': 'amos',
+                         'password': 'password'})
+    csrf = testapp.get('/add_smack').html.find_all("input")
+    csrf = csrf[0].attrs['value']
+    html_user = testapp.get('/userlist',
+                            params={'csrf_token': csrf},
+                            status=200).html.find_all('td')
+    assert '<td colspan="2"><b>True</b></td>' == str(html_user[1])
```

Comment:
Why do you need a CSRF token to view the user list? Also, shouldn't part of your assert be that you see one of the users you expect in the page?